### PR TITLE
Struct generation by importing JSON

### PR DESCRIPTION
**Pull request by keizer619**
This feature will add a button called "Import from JSON" in the Struct Definition view.

screen shot 2017-10-27 at 10 36 26 am

User can provide a sample JSON on pop up dialog box 

screen shot 2017-10-27 at 10 37 11 am

Upon importing Struct fields will be generated from the JSON 

screen shot 2017-10-27 at 10 37 26 am

### CODE REVIEW

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx [hunk 1]
```diff
@@ -187,6 +188,67 @@ class StructNode extends React.Component {
     }
 
     /**
+     * Open JSON import dialog box and pass callback
+     */
+    onClickJsonImport() {
+        const onImport = (json) => {
+            let refExpr = TreeBuilder.build(FragmentUtils.parseFragment(FragmentUtils.createExpressionFragment(json)));
+            let currentValue;
+            this.props.model.setFields([], true);
```
Comment (maheeka) [inline modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx]: Should we check for errors in building the tree. If there is an error, we shouldn't make fields empty.

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx [hunk 2]
```diff
@@ -187,6 +188,67 @@ class StructNode extends React.Component {
...
+            refExpr.variable.initialExpression.keyValuePairs.forEach((ketValPair) => {
```
Comment (maheeka) [inline modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx]: Also do a isRefExpr check if we are not validating the json content to be json in the text area.

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx [hunk 3]
```diff
@@ -187,6 +188,67 @@ class StructNode extends React.Component {
...
+            },
+        });
+    }
+
+    /**
+     * Check given value is an Integer
+     * @param  {Number} val value to check
+     * @return {Boolean}  is Intger
+     */
+    isInt(val) {
```
Comment (maheeka) [inline modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx]: Can we avoid this individual type check? Or should it be for all primitive types? Not sure what we are doing here.

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx (hunk 2 above)
Comment (keizer619) [reply modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx]: JSON will be validated from the ACE editor. Import button will be only enabled if its a valid JSON

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx (hunk 3 above)
Comment (keizer619) [reply modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx]: Sample JSON may contain 23 or 23.5  to figure out if its a Integer or Float to generate struct types these checks are used.

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx (hunk 1 above)
Comment (keizer619) [reply modules/web/js/ballerina/diagram2/views/default/components/nodes/struct-node.jsx]: As JSON is validated from ACE editor mode, there should not be errors when parsing

**Pull request merged**